Dockerfile

```diff
@@ -1,22 +1,49 @@
-FROM tomcat:8.5-jdk8-openjdk
-MAINTAINER geosolutions<info@geo-solutions.it>
+FROM alpine AS mother
+LABEL maintainer="Alessandro Parma<alessandro.parma@geo-solutions.it>"
+ARG MAPSTORE_WEBAPP_SRC="./.placeholder"
+ADD "${MAPSTORE_WEBAPP_SRC}" "/mapstore/"
+
+COPY ./docker/* /mapstore/docker/
+WORKDIR /mapstore
+
+FROM tomcat:9-jdk11-openjdk
 
 # Tomcat specific options
 ENV CATALINA_BASE "$CATALINA_HOME"
-ENV JAVA_OPTS="${JAVA_OPTS}  -Xms512m -Xmx512m -XX:MaxPermSize=128m"
+ENV MAPSTORE_WEBAPP_DST="${CATALINA_BASE}/webapps"
+ENV INITIAL_MEMORY="512m"
+ENV MAXIMUM_MEMORY="512m"
+ENV JAVA_OPTS="${JAVA_OPTS}  -Xms${INITIAL_MEMORY} -Xmx${MAXIMUM_MEMORY} -XX:MaxPermSize=128m"
+ENV GEOSTORE_OVR_OPT="-Dgeostore-ovr=${CATALINA_BASE}/conf/${OVR} \
+                      -Duser.timezone=UTC"
+ARG TOMCAT_EXTRAS=false
+ARG OVR=""
 
 # Optionally remove Tomcat manager, docs, and examples
-ARG TOMCAT_EXTRAS=false
 RUN if [ "$TOMCAT_EXTRAS" = false ]; then \
-      find "${CATALINA_BASE}/webapps/" -delete; \
+      find "${MAPSTORE_WEBAPP_DST}" -delete; \
+    fi
+
+COPY --from=mother "/mapstore/mapstore.war" "${MAPSTORE_WEBAPP_DST}/mapstore.war"
+COPY --from=mother "/mapstore/docker" "${CATALINA_BASE}/docker/"
+
+WORKDIR ${CATALINA_BASE}
+#RUN chmod +x ./docker/wait-for-postgres.sh
+#COPY ./docker/wait-for-postgres.sh ${CATALINA_BASE}/
+# adding the env
+RUN if [ "${OVR}" = "geostore-datasource-ovr.properties" ]; then \
+        cp ${CATALINA_BASE}/docker/geostore-datasource-ovr.properties ${CATALINA_BASE}/conf; \
     fi
 
-# Add war files to be deployed
-COPY docker/*.war "${CATALINA_BASE}/webapps/"
+RUN apt-get update \
+    && apt-get install --yes postgresql-client-11 \
```

Comment:
please split the PG client installation and packages cleanup in two distinct RUN steps and make the PostrgeSQL client version number an ENV defined at the beginning of the dockerfile so that it is easy to change it

Reply:
PG client split and ENV added in this commit https://github.com/geosolutions-it/MapStore2/pull/6490/commits/81f252222042c06992b67e8602977de593b31564